### PR TITLE
add info in doc on how to facet with cartopy

### PR DESCRIPTION
**Pull request by vnoel**
This change explains
- how to pass projection arguments to faceted subplots
- how to access the Axes of the created subplots (not relevant only to cartopy), relevant to issue #1202

### CODE REVIEW

File: doc/plotting.rst [hunk 1]
```diff
@@ -446,6 +446,16 @@ Here is the resulting image:
 
 .. image:: examples/cartopy_example.png
 
+When faceting on maps, the projection can be transferred to the plot function using the ``subplot_kws`` keyword. Axes for the subplots created by faceting are accessible in the object returned by plot:
+
+.. ipython:: python
+
+    p = time_maps.plot(transform=ccrs.PlateCarree(), col='time', subplot_kws={'projection':ccrs.PlateCarree()})
```
Comment (shoyer) [inline doc/plotting.rst]: please add a space after `:` per pep8

**Comment by fmaussion** [comment]
Thanks @vnoel ! 

Unfortunately this won't render well in the docs since we don't import cartopy (yet) (the example above yours isn't really compiling so it's not importing cartopy). I'll merge this tomorrow and have a look at how to render this properly in a separate PR

On a related note, I agree that we should make a plot gallery (https://github.com/pydata/xarray/issues/1061). I've tried it out with salem and it looks nice (http://salem.readthedocs.io/en/latest/auto_examples/index.html). I'll get back to this when I'll find some time...

**Comment by avatar101** [comment]
@vnoel I'm trying to plot coastline using facet. I tried to give `ax.coastline()` as a subplot_kws argument but that didn't work:

    # ds_v_test is my Dataset containing V as variable

`p = ds_v_test.sel(time=slice('2012-12-01', '2012-12-02 18:00'),\
              lat= slice(-90,0)).V.squeeze().plot.pcolormesh(figsize=(16, 12), col='time', col_wrap=2, levels=16 ,\
                                                             cbar_kwargs={'label':'meridional v (m/s)'}, subplot_kws={'projection':ccrs.PlateCarree(),\
                                                                                                                     'ax':ax.coastlines()})`
It gives me 

> AttributeError: Unknown property ax

At the moment, I can get around it by using this approach:

 `for ax in p.axes.flat:
     ax.coastlines()`

> INSTALLED VERSIONS
>------------------

> commit: None
> python: 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
> [GCC 7.3.0]
> python-bits: 64
> OS: Linux
> OS-release: 4.15.0-45-generic
> machine: x86_64
> processor: x86_64
> byteorder: little
> LC_ALL: None
> LANG: en_US.UTF-8
> LOCALE: en_US.UTF-8
> libhdf5: 1.10.4
> libnetcdf: 4.6.2
> 
> xarray: 0.12.0
> pandas: 0.24.2
> numpy: 1.16.2
> scipy: 1.2.1
> netCDF4: 1.4.2
> pydap: None
> h5netcdf: None
> h5py: None
> Nio: 1.5.5
> zarr: None
> cftime: 1.0.3.4
> nc_time_axis: None
> PseudonetCDF: None
> rasterio: None
> cfgrib: 0.9.6.1.post1
> iris: None
> bottleneck: None
> dask: None
> distributed: None
> matplotlib: 3.0.3
> cartopy: 0.17.0
> seaborn: 0.9.0
> setuptools: 40.8.0
> pip: 19.0.3
> conda: None
> pytest: None
> IPython: 7.3.0
> sphinx: None
> 
> ​

**Comment by dcherian** [comment]
Yes you'll have to call `ax.coastlines()` on every axis you want a coastline on. The `subplot_kws` is passed down to `plt.subplots` (https://matplotlib.org/api/_as_gen/matplotlib.pyplot.subplots.html).

You should be able to do:
``` python
g = ds_v_test.plot(...)
g.map(plt.gca().coastlines())
```

**Comment by avatar101** [comment]
@dcherian Thanks a lot for providing an example with another approach

**Pull request merged**